linera-views/src/memory.rs

```diff
@@ -193,84 +201,109 @@ impl WritableKeyValueStore<MemoryStoreError> for MemoryStore {
+            Arc::new(RwLock::new(map))
+        });
         let map = store.clone();
         let namespace = namespace.to_string();
+        let root_key = root_key.to_vec();
         Ok(MemoryStore {
             map,
             max_stream_queries,
             namespace,
+            root_key,
             kill_on_drop,
         })
     }
 
-    fn sync_list_all(namespace_memory_store: &NamespaceMemoryStore) -> Vec<String> {
-        namespace_memory_store.keys().cloned().collect::<Vec<_>>()
+    fn sync_list_all(memory_stores: &MemoryStores) -> Vec<String> {
+        memory_stores.namespaces.iter().cloned().collect::<Vec<_>>()
     }
 
-    fn sync_exists(namespace_memory_store: &NamespaceMemoryStore, namespace: &str) -> bool {
-        let namespace = namespace.to_string();
-        namespace_memory_store.contains_key(&namespace)
+    fn sync_exists(memory_stores: &MemoryStores, namespace: &str) -> bool {
+        memory_stores.namespaces.contains(namespace)
     }
 
-    fn sync_create(namespace_memory_store: &mut NamespaceMemoryStore, namespace: &str) {
-        let namespace = namespace.to_string();
-        let map = MemoryStoreMap::new();
-        let map = Arc::new(RwLock::new(map));
-        namespace_memory_store.insert(namespace, map);
+    fn sync_create(memory_stores: &mut MemoryStores, namespace: &str) {
+        memory_stores.namespaces.insert(namespace.to_string());
     }
 
-    fn sync_delete(namespace_memory_store: &mut NamespaceMemoryStore, namespace: &str) {
+    fn sync_delete(memory_stores: &mut MemoryStores, namespace: &str) {
         let namespace = namespace.to_string();
-        namespace_memory_store.remove(&namespace);
+        memory_stores.namespaces.remove(&namespace);
+        let mut pair_removes = Vec::new();
+        for key in memory_stores.stores.keys() {
+            if key.0 == namespace {
+                pair_removes.push(key.clone());
+            }
+        }
+        for pair in pair_removes {
+            memory_stores.stores.remove(&pair);
+        }
     }
 
     /// Create a memory store if one is missing and otherwise connect with the existing one
     fn sync_maybe_create_and_connect(
         config: &MemoryStoreConfig,
         namespace: &str,
+        root_key: &[u8],
         kill_on_drop: bool,
     ) -> Result<Self, MemoryStoreError> {
-        let mut namespace_memory_store = MEMORY_STORES.lock().expect("lock should not be poisoned");
-        if !MemoryStore::sync_exists(&namespace_memory_store, namespace) {
-            MemoryStore::sync_create(&mut namespace_memory_store, namespace);
+        let mut memory_stores = MEMORY_STORES.lock().expect("lock should not be poisoned");
+        if !MemoryStore::sync_exists(&memory_stores, namespace) {
+            MemoryStore::sync_create(&mut memory_stores, namespace);
         }
-        MemoryStore::sync_connect(&namespace_memory_store, config, namespace, kill_on_drop)
+        MemoryStore::sync_connect(
```

Comment:
Why the prefix `sync_` on all these methods?

Reply:
It turns out that the functionality of the memory store do not require the code to be async. This is due to the locks being `sync` instead of `async`. This is nice as it allows the creation of test contexts without having async code which is quite practical.

To reflect this, the corresponding functions use the `sync_` prefix to distinguish them from other functions.